### PR TITLE
Add missing parameters to DecorationImage

### PR DESCRIPTION
**Pull request by nt4f04uNd**
Adding missing parameters to the DecorationImage:
* opacity
* filterQuality
* invertColors
* isAntiAlias

In `RawImage` the `opacity` is actually an `Animation`, but since that's a rendering library, we cannot use it here. It should be a responsibility of the creator to repaint the decoration if value changes

Fixes https://github.com/flutter/flutter/issues/88436

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/lib/src/painting/decoration_image.dart [hunk 1]
```diff
@@ -142,6 +146,35 @@ class DecorationImage {
...
   final double scale;
 
+  /// If non-null, the value is multiplied with the opacity of each image
+  /// pixel before painting onto the canvas.
+  ///
+  /// This is more efficient than using [Opacity] or [FadeTransition] to
+  /// change the opacity of an image.
+  final double opacity;
+
+  /// Used to set the filterQuality of the image
```
Comment (craiglabenz) [inline packages/flutter/lib/src/painting/decoration_image.dart:156]: nit: `image.`

(missing period)

Comment (craiglabenz) [reply packages/flutter/lib/src/painting/decoration_image.dart:156]: Also, I believe the style guide calls for a newline between this and following docstring content, as your other docstrings do :)

File: packages/flutter/lib/src/painting/decoration_image.dart [hunk 2]
```diff
@@ -142,6 +146,35 @@ class DecorationImage {
...
+  /// Use the "low" quality setting to scale the image, which corresponds to
+  /// bilinear interpolation, rather than the default "none" which corresponds
+  /// to nearest-neighbor.
+  final FilterQuality filterQuality;
+
+  /// Whether the colors of the image are inverted when drawn.
+  ///
+  /// inverting the colors of an image applies a new color filter to the paint.
```
Comment (craiglabenz) [inline packages/flutter/lib/src/painting/decoration_image.dart:164]: nit: `Inverting`

(capitalization)

File: packages/flutter/lib/src/painting/decoration_image.dart [hunk 3]
```diff
@@ -189,7 +239,13 @@ class DecorationImage {
         '$repeat',
       if (matchTextDirection)
         'match text direction',
-      'scale: $scale',
+      'scale $scale',
+      'opacity $opacity',
+      '$filterQuality',
+      if (invertColors)
+        'invert colors',
```
Comment (craiglabenz) [inline packages/flutter/lib/src/painting/decoration_image.dart:246]: I wonder if this should say `invertColors: true` 🤔

File: packages/flutter/test/painting/decoration_test.dart [hunk 4]
```diff
@@ -316,19 +320,30 @@ void main() {
...
-    // ignore: avoid_dynamic_calls
-    expect(call.positionalArguments[3].colorFilter, colorFilter);
-    // ignore: avoid_dynamic_calls
-    expect(call.positionalArguments[3].filterQuality, FilterQuality.low);
+    final Paint paint = call.positionalArguments[3] as Paint;
+    expect(paint.colorFilter, colorFilter);
+    expect(paint.color, const Color(0x7F000000)); // 0.5 opacity
+    expect(paint.filterQuality, FilterQuality.high);
+    expect(paint.isAntiAlias, true);
+    expect(paint.invertColors, true);
```
Comment (craiglabenz) [inline packages/flutter/test/painting/decoration_test.dart:328]: Nice :)

**Review by craiglabenz — APPROVED** [review]
LGTM pending a few punctuation issues in the new docstrings.

File: packages/flutter/lib/src/painting/decoration_image.dart (hunk 3 above)
Comment (nt4f04uNd) [reply packages/flutter/lib/src/painting/decoration_image.dart:246]: I followed the style above with matchTextDirection. Should we change that for both or just for inverting colors, same for `isAntiAlias`?

File: packages/flutter/lib/src/painting/decoration_image.dart (hunk 2 above)
Comment (nt4f04uNd) [reply packages/flutter/lib/src/painting/decoration_image.dart:164]: Most of the docs are copied from RawImage. I also noticed those things, but wanted to make them in the separate PR, I guess I will change them in this one then

**Comment by craiglabenz** [comment]
Oh yes, so you did. Never mind then, don't bother with that change. It was
a bad idea on my account!

File: packages/flutter/lib/src/painting/decoration_image.dart (hunk 3 above)
Comment (craiglabenz) [reply packages/flutter/lib/src/painting/decoration_image.dart:247]: Oh yes, so you did. Never mind then, don't bother with that change. It was
a bad idea on my account!

**Comment by nt4f04uNd** [comment]
I guess one of the tests fails on web. Is there a way for me to run it locally to see what's wrong?

**Comment by craiglabenz** [comment]
I suspect that test is flakey, so I'm rerunning it.

**Comment by nt4f04uNd** [comment]
I don't think so, unfortunately trying to run it locally hangs of forever on compilation

**Comment by nt4f04uNd** [comment]
I believe it's because `invertColors` is not working on web. What should I do?

**Comment by craiglabenz** [comment]
It looks like the tests in `test/painting/decoration_test.dart` are failing in the web build despite passing when run normally. I'm not an expert on that layer, so I'm trying to track it down.

**Pull request merged**